server/py_files/Spacy.py

```diff
@@ -1,22 +1,37 @@
 import re
 
 import spacy
-import en_core_web_sm
-nlp = en_core_web_sm.load()
+import en_core_web_lg
+nlp = en_core_web_lg.load()
 nlp.max_length = 2000000 # todo check this limit
 
-import time
-
 class Spacy(object):
-    def anonymize(self, t):
+    def anonymize_token(self, t):
         if t.like_email: # todo bring in Human Names as well
             return 'X'*len(t.text)
         return t.text
 
-    # todo need a has_phone_number instead
-    def is_phone_number(self, text):
-        phone_regex = re.compile('(?:(?:\+?1\s*(?:[.-]\s*)?)?(?:(\s*([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9])\s*)|([2-9]1[02-9]|[2-9][02-8]1|[2-9][02-8][02-9]))\s*(?:[.-]\s*)?)([2-9]1[02-9]|[2-9][02-9]1|[2-9][02-9]{2})\s*(?:[.-]\s*)?([0-9]{4})')
-        return phone_regex.match(text)
+    def anonymize_name(self, text):
+        doc = nlp(text)
+        spans = list(doc.ents)
+
+        for span in spans:
+            span.merge()
+
+        for person in filter(lambda w: w.ent_type_ == 'PERSON', doc):
```

Comment:
@tzunun : if all we needed was `ent_type_`, then I think that is available on the token itself (while tokenizing)

refer https://spacy.io/api/token => search for `https://spacy.io/api/token`

right now in `anonymize_name()` we are ending up tokenizing twice (costly operation) -- as we are already doing a `nlp(str(text))` inside the `tokenize()` function

Instead of that, similar to `t.like_email` inside `anonymize_token()`, we can do a `t. ent_type_  == 'PERSON'` -- and using 1 single tokenizing step, get everything we want